File: codewars/05-build-tower/tower.py
def tower_builder(n_floors):
    stars = 1
    spaces = n_floors-1
    a = []
    for i in range(n_floors):
        a.append(' ' * spaces + '*' * stars + ' ' * spaces)
        print(' ' * spaces + '*'*(stars))
        spaces -= 1
        stars += 2
    print(a)
    return a

def tower_builder2(n):
    return [("*" * (i*2-1)).center(n*2-1) for i in range(1, n+1)]

def tower_builder3(n_floors):
    if n_floors == 0:
        return []
        
    count = 1
    result = []
    for i in range(1, n_floors + 1):
        stars = '*' * (2 * i - 1)
        space = ' ' * (n_floors - i)
        result.append(space + stars + space)
    return result

def tower_builder(n):
    return [" " * (n - i - 1) + "*" * (2*i + 1) + " " * (n - i - 1) for i in range(n)]

if __name__ == '__main__':
    tower_builder(0)

File: codewars/05-build-tower/tower.cpp
#include <iostream>
#include <string>
#include <vector>

std::vector<std::string> towerBuilder(unsigned nFloors) {
  std::vector<std::string> result;
  std::string stars = "*";
  std::string spaces = "";
  int nSpaces = nFloors - 1;

  for (int j = 0; j < nFloors; j++) spaces += " ";

  for (int i = 0; i < nFloors; i++) {
    spaces = "";
    for (int k = 0; k < nSpaces; k++) spaces += " ";
    nSpaces -= 1;
    result.push_back(spaces + stars + spaces);
    stars += "**";
  }
  return result;
}

std::vector<std::string> towerBuilder2(unsigned nFloors) {
  std::vector<std::string> vect;
  for (unsigned i = 0, k = 1; i < nFloors; i++, k += 2)
    vect.push_back(std::string(nFloors - i - 1, ' ') + std::string(k, '*') +
                   std::string(nFloors - i - 1, ' '));
  return {vect};
}

std::vector<std::string> towerBuilder3(unsigned n) {
  std::vector<std::string> tower = {};
  for (int i = 1; i <= n; i++) {
    std::string stars(2 * i - 1, '*');
    std::string spaces(n - i, ' ');
    tower.push_back(spaces + stars + spaces);
  }
  return tower;
}

int main() {
  towerBuilder(6);
  for (std::string i : towerBuilder(6)) std::cout << i << "\n";

  return 0;
}